components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -1147,6 +1147,87 @@ void JsonRpcService::OnGetERC20TokenAllowance(
   std::move(callback).Run(args->at(0), mojom::ProviderError::kSuccess, "");
 }
 
+void JsonRpcService::GetERC20TokenBalances(
+    const std::vector<std::string>& token_contract_addresses,
+    const std::string& user_address,
+    const std::string& chain_id,
+    GetERC20TokenBalancesCallback callback) {
+  const auto& balance_scanner_contract_addresses =
+      GetEthBalanceScannerContractAddresses();
+  if (!base::Contains(balance_scanner_contract_addresses, chain_id)) {
+    std::move(callback).Run(
+        {}, mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+  const auto& balance_scanner_contract_address =
+      balance_scanner_contract_addresses.at(chain_id);
+
+  if (token_contract_addresses.empty() || user_address.empty() ||
+      balance_scanner_contract_address.empty()) {
+    std::move(callback).Run(
+        {}, mojom::ProviderError::kInvalidParams,
+        l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+    return;
+  }
+
+  absl::optional<std::string> calldata =
+      balance_scanner::TokensBalance(user_address, token_contract_addresses);
```

Comment:
JsonRpcService::GetERC20 `TokenBalances`
balance_scanner:: `TokensBalance`
I'd suggest we have one form everywhere.

Reply:
My thinking was that `TokensBalance` reflects the function of the contract https://etherscan.io/address/0x08A8fDBddc160A7d5b957256b903dCAb1aE512C5#readContract, whereas JsonRpcService::GetERC20TokenBalances is mostly a pluralization of the existing JsonRpcService::GetERC20TokenBalance method (which actually calls the `BalanceOf` function).  Does that change your thinking at all? If not I'll probably rename GetERC20TokenBalances.

Reply:
I'm thinking of it as we have many tokens and get balance for each instead of many balances for one token, so `tokensbalance`, but I'm not native English speaker :)
Actually I'm okay with any way you prefer, just wanted to make sure it is not a typo.

Reply:
Btw contract and lib(from comment) support both `tokensbalance` and `tokenbalances`
https://github.com/MyCryptoHQ/eth-scan#gettokenbalancesprovider-addresses-token-options
https://github.com/MyCryptoHQ/eth-scan#gettokensbalanceprovider-address-tokens-options
Mixing them may confuse someone

Reply:
>  If not I'll probably rename GetERC20TokenBalances.

^ this part was a typo, I meant GetERC20TokensBalance (to match the contract function name).  But yes I agree it's confusing, and the naming in the contract function could be better.  If we wanted to support the other method I'd name it GetBalancesForERC20Token.  GetERC20TokenBalances feels more appropriate just because most of the APIs we have are in context of doing stuff for a user.

I find TokensBalance confusing in any context tbh